terraform/modules/hub/matomo.tf

```diff
@@ -0,0 +1,536 @@
+locals {
+  volume_name = "shared_www"
+
+  log_format = <<-LOG_FORMAT
+  log_format  matomo  '{"ip": "$remote_addr",'
+                       '"http_x_forwarded_for": "$http_x_forwarded_for",'
+                       '"host": "$host",'
+                       '"path": "$request_uri",'
+                       '"status": "$status",'
+                       '"referrer": "$http_referer",'
+                       '"user_agent": "$http_user_agent",'
+                       '"length": $bytes_sent,'
+                       '"generation_time_milli": $request_time,'
+                       '"userid": "$remote_user",'
+                       '"request": "$request",'
+                       '"msec": "$msec",'
+                       '"method": "$request_method",'
+                       '"content_type": "$content_type",'
+                       '"date": "$time_iso8601"}';
+
+  access_log /dev/stdout matomo;
+LOG_FORMAT
+
+
+  nginx_log_format_base64 = base64encode(local.log_format)
+}
+
+resource "aws_kms_key" "matomo" {
+  description = "used to encrypt secrets in parameter store for matomo"
+}
+
+# resource "aws_kms_alias" "matomo" {
+#   name          = "alias/platform-web"
+#   target_key_id = aws_kms_key.matomo_web.key_id
+# }
+
+resource "aws_lb" "matomo" {
+  name               = "matomo" ## TODO: namespace
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.matomo_lb.id]
+  subnets            = aws_subnet.ingress.*.id
+  idle_timeout       = 300
+}
+
+# resource "aws_lb_listener" "matomo_http" {
```

Comment:
why is this commented out?

Reply:
copied from old world but don't think it's needed ... will find out in the next story